dndtroops-persistence/src/main/java/cz/muni/fi/pa165/dndtroops/entities/Troop.java

```diff
@@ -1,7 +1,128 @@
 package cz.muni.fi.pa165.dndtroops.entities;
 
+import javax.persistence.*;
+import java.util.Objects;
+
+/*
+
+Entity class for entity Troop.
+Creted by: Vojtech Duchon (UCO:410007)
+
+ */
+
+
+@Entity
+@Table (name = "troop")
 public class Troop {
 
-    // TODO
+
+    @Id
+    @GeneratedValue (strategy = GenerationType.IDENTITY)
+    @Column (name= "id")
+    private long id;
+
+    @Column(name= "name")
+    private String name;
+
+    @Column (name= "mission")
+    private String mission;
+
+    @Column (name= "goldenMoney")
+    private long goldenMoney;
+
+    /*
+    No argument constructor.
+     */
+
+    public Troop(){
+
+    }
+
+    /*
+    Constructor with specified attributes.
+     */
+
+
+    public Troop(long id, String name, String mission, long goldenMoney)
+    {
+        this.id = id;
+        this.name=name;
+        this.mission=mission;
+        this.goldenMoney=goldenMoney;
+
+    }
+
+    /*
+    Override of default Object.equals() method
+     */
+
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
I think this is not a business key. When I have two same troops, but only with the different amount of money, they should be equal.

Also, a part of the business key is NOT entity's ID.